src/main/scala/com/clevercloud/warp10client/Runner.scala

```diff
@@ -48,20 +50,31 @@ object Runner {
     import warpClientContext._
 
     Flow[Try[HttpResponse]].flatMapConcat {
-      case Success(httpResponse) => {
-        if (httpResponse.status == StatusCodes.OK) {
-          Source.future(
-            WarpClientUtils.readAllDataBytes(httpResponse.entity.dataBytes)
-          )
+      case Success(httpResponse) =>
+        Source.future(if (httpResponse.status == StatusCodes.OK) {
+          WarpClientUtils.readAllDataBytes(httpResponse.entity.dataBytes)
+        } else if (httpResponse.status == StatusCodes.InternalServerError) {
+          // In case of 500, we try to read headers which are human-readable and fallback on response body which is HTML
+          val error: Option[String] = httpResponse.headers.findLast(_.is("x-warp10-error-message")).map(_.value)
+          val line: Option[Int] = httpResponse.headers.findLast(_.is("x-warp10-error-line")).map(_.value).flatMap { v =>
+            try {
+              Some(v.toInt)
+            } catch {
+              case _: Exception => None
+            }
+          }
+
+          error match
+            case Some(err) => Future.successful(throw WarpException(err, line))
+            case None =>
+              WarpClientUtils.readAllDataBytes(httpResponse.entity.dataBytes).map(WarpException(_, line)).map(throw _)
```

Comment:
Can you add a test to ensure the try handles the throw well? Or use either is you prefer (change the method output).